website/user_activity.py

```diff
@@ -0,0 +1,63 @@
+
+import os
+from flask import request, session
+
+# import utils
+from config import config
+from website.auth import requires_login, is_teacher
+
+from .database import Database
+from .website_module import WebsiteModule, route
+
+from website import s3_logger
+
+cookie_name = config["session"]["cookie_name"]
+invite_length = config["session"]["invite_length"] * 60
+
+
+# Add LOG_USER_ACTIVITY=True to your env. to log the data for testing
+write_to_file = os.getenv("LOG_USER_ACTIVITY")
+logger = s3_logger.S3Logger(name="activity", config_key="s3-activity-logs",
+                            write_to_file=write_to_file,)
```

Comment:
How about the following:

```
if utils.is_herokuo(): 
    logger = s3_logger.S3Logger(name="activity", config_key="s3-activity-logs")
elif write_to_file:
  logger = s3_logger.FileLogger(filename="user_activity.json")
else:
  logger = s3_logger.NullLogger()
```

?

Reply:
Don't mind, it's just more code everywhere we use it